### PR TITLE
refactor potions and leather armor

### PR DESCRIPTION
**Pull request by nstringham**
note: this PR contains #520 so you should probably merge that one first

this PR performs some refactors to leather armor textures then it migrates potions from static images to a system like leather armor

Note for server maintainer:
- this PR adds a /potions route which acts like /leather and should be added to configs for Nginx, Cloudflare, etc.
- this PR changes the files naming scheme for leather armor inside cache. To prevent storing dead files I recommend deleting the entire cache directory after deploying this PR.

before:
image

after:
image

1.16:
image

1.8:
image

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -387,26 +387,52 @@ async function main() {
...
-    let file;
+    const { type, color } = req.params;
 
-    if (!["boots", "leggings", "chestplate", "helmet"].includes(req.params.type)) {
+    if (!["boots", "leggings", "chestplate", "helmet"].includes(type)) {
       throw new Error("invalid armor type");
     }
 
-    const { type } = req.params;
+    if (!/^[0-9a-fA-Z]{6}$/.test(color)) {
```
Comment (dukio) [inline src/app.js:396]: Shouldn't this be a-fA-F?

File: src/app.js [hunk 2]
```diff
@@ -387,26 +387,52 @@ async function main() {
...
+
+    res.setHeader("Cache-Control", `public, max-age=${CACHE_DURATION}`);
+    res.contentType("image/png");
+    res.send(file);
+  });
+
+  app.all("/potion/:type/:color", cors(), async (req, res) => {
+    const { type, color } = req.params;
+
+    if (!/^[0-9a-fA-Z]{6}$/.test(color)) {
```
Comment (dukio) [inline src/app.js:424]: Again, a-fA-F?

File: src/app.js (hunk 1 above)
Comment (nstringham) [reply src/app.js]: 72d00e0581adab835fb6a0c2d06e3e1382430d7f

File: src/app.js (hunk 2 above)
Comment (nstringham) [reply src/app.js]: 72d00e0581adab835fb6a0c2d06e3e1382430d7f

**Comment by nstringham** [comment]
thanks @dukio

**Comment by nstringham** [comment]
This PR has a vulnerability I need to fix can someone with write access convert it to a draft for me

**Comment by nstringham** [comment]
Fixed the potential vulnerability ready for review again

also added a Note for server maintainer to the original comment

**Review by MartinNemi03 — APPROVED** [review]

**Review by metalcupcake5 — APPROVED** [review]

**Pull request merged**